### PR TITLE
Add snippet register page

### PR DESCRIPTION
**Pull request by festinalente91**
- book search
- image upload
- image ocr
- snippet register
* npm install is required.
* schema update required.

### CODE REVIEW

File: app/components/create/book.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import React, { Component, PropTypes } from 'react';
+import { View, TextInput, Text, TouchableOpacity, Image } from 'react-native';
+import Styles from './styles';
+
+export class Book extends Component {
+
```
Comment (ace0625) [inline app/components/create/book.js:6]: Delete empty line here

File: app/components/create/book.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+    };
+  }
+
+  render() {
+    let title = this.props.book.title;
```
Comment (ace0625) [inline app/components/create/book.js]: I think you can do this task in constructor or componentDidMount.

**Review by ace0625 — APPROVED** [review]

File: app/components/create/bookSearchBox.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    return (
+      <View>
+        {this.state.isSelected && <TagList onCancelTag={this.onCancelTag} data={this.state.tagList}></TagList>}
+        {!this.state.isSelected && <BookSearchBar searchInputEventEmitter={this.searchInputEventEmitter} /> }
+        {!this.state.isSelected &&
+          <BookSearchResult
+            onBookSelected={this.onBookSelected}
+            onpressBook={this.props.onpressBook}
+            searchInput$={this.searchInput$} />
+
```
Comment (SarahhhC) [inline app/components/create/bookSearchBox.js]: Delete line

File: package.json [hunk 4]
```diff
@@ -7,14 +7,19 @@
     "test": "eslint app/**/*.js && jest"
   },
   "dependencies": {
+    "querystring": "^0.2.0",
     "react": "15.4.1",
     "react-native": "0.39.2",
+    "react-native-aws3": "0.0.4",
     "react-native-camera": "^0.4",
-    "react-native-flip-card": "^3.3.1",
```
Comment (SarahhhC) [inline package.json:13]: Why deleted???

File: app/mutations/snippet.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    }), {
+      onSuccess: (res) => {
+        resolve(res);
+      },
+      onFailure: (transaction) => {
+        reject(transaction.getError().message);
+      }
+    });
+  });
+}
```
Comment (SarahhhC) [inline app/mutations/snippet.js]: Add one line~

File: app/mutations/bookSnippet.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    }), {
+      onSuccess: (res) => {
+        resolve(res);
+      },
+      onFailure: (transaction) => {
+        reject(transaction.getError().message);
+      }
+    });
+  });
+}
```
Comment (SarahhhC) [inline app/mutations/bookSnippet.js]: Add one line.

File: app/mutations/snippet.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    return Relay.QL`
+        fragment on createSnippetPayload {
+            user {
+                email
+            }
+        }
+    `;
+  }
+
+  getConfigs() {
```
Comment (SarahhhC) [inline app/mutations/snippet.js:35]: What is difference between `getFatQuery` and `getConfigs `?

Comment (ace0625) [reply app/mutations/snippet.js:35]: https://facebook.github.io/relay/docs/guides-mutations.html
This might help.

Comment (SarahhhC) [reply app/mutations/snippet.js:35]: Thanks!!

File: package.json (hunk 4 above)
Comment (festinalente91) [reply package.json:13]: by accident. sry
oh! Seems I didn't deleted it :)

File: app/components/create/book.js (hunk 1 above)
Comment (festinalente91) [reply app/components/create/book.js:6]: deleted.

File: app/mutations/snippet.js (hunk 5 above)
Comment (festinalente91) [reply app/mutations/snippet.js]: added.

File: app/mutations/bookSnippet.js (hunk 6 above)
Comment (festinalente91) [reply app/mutations/bookSnippet.js]: added.

File: app/components/create/book.js [hunk 8]
```diff
@@ -7,13 +7,12 @@ export class Book extends Component {
   constructor(props) {
     super(props);
     this.state = {
+      thumbnail: this.props.book.thumbnail ?
+        this.props.book.thumbnail.replace('http', 'https') : this.props.book.thumbnail
```
Comment (SarahhhC) [inline app/components/create/book.js]: Nice!

File: app/components/create/createSnippet.js [hunk 9]
```diff
@@ -92,7 +92,8 @@ class CreateSnippet extends Component {
                     isbn: this.state.book.isbn,
                     author: this.state.book.author,
                     category: this.state.book.category,
-                    thumbnail: this.state.book.thumbnail,
+                    thumbnail: this.state.book.thumbnail ?
+                      this.state.book.thumbnail.replace('http', 'https') : this.state.book.thumbnail,
```
Comment (SarahhhC) [inline app/components/create/createSnippet.js]: Good!

**Pull request merged**